File: @LV1/1.4/modulo/modulo.cpp
#include <cstdio>

int main() {
    int visited[42] = {0};
    int n, cnt = 0;

    while (scanf("%d", &n) == 1) {
        if (visited[n % 42] == 0) {
            visited[n % 42] = 1;
            cnt++;
        }
    }
    printf("%d\n", cnt);
}

File: @LV1/1.4/modulo/modulo.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	reader := bufio.NewScanner(os.Stdin)
	visited := make([]int, 42)
	cnt := 0

	for reader.Scan() {
		n, _ := strconv.Atoi(reader.Text())
		n %= 42

		if visited[n] == 0 {
			cnt++
			visited[n] = 1
		}
	}
	fmt.Println(cnt)
}
